src/strdup.c

```diff
@@ -31,20 +32,19 @@ rcutils_strdup(const char * str, rcutils_allocator_t allocator)
 {
   RCUTILS_CAN_RETURN_WITH_ERROR_OF(NULL);
 
-  if (NULL == str) {
-    return NULL;
-  }
-  return rcutils_strndup(str, strlen(str), allocator);
+  return rcutils_strndup(str, SIZE_MAX, allocator);
```

Comment:
Does this have to be changed into `SIZE_MAX`? Could you give us the problem case?

Reply:
Using `strlen()` iterates over the string unnecessarily.